src/main/java/com/hcl/appscan/jenkins/plugin/scanners/StaticAnalyzer.java

```diff
@@ -222,6 +225,9 @@ public Map<String,String> getProperties(VariableResolver<String> resolver) {
             if(m_scanSpeed!=null && !m_scanSpeed.isEmpty() && getHasOptions()) {
                 properties.put(SCAN_SPEED, m_scanSpeed);
             }
+            if(getRescan() && getScanId()!=null && !getScanId().isEmpty() ){
+                properties.put(CoreConstants.SCAN_ID,getScanId());
+            }
```

Comment:
I don't think we should allow the scan to proceed if the user selected the rescan option, but didn't provide a scan id. This method (getProperties()) is an override of the method in the Scanner class that throws an AbortException. We should add the @Override annotation to this method and the throws declaration and throw an AbortException if the user selected to run a rescan, but didn't provide a scan id.
On a side note, we should probably add a method to the base Scanner class for checking if a string is null or empty, since that check is done in a lot of places in both the StaticAnalyzer and DynamicAnalyzer classes. For example:
```
protected boolean isNullOrEmpty(String string) {
    return string != null && !string.trim().isEmpty();
}
```

Reply:
Sure, Matt. Added "isNullOrEmpty()" method in the scanner class.